components/QuantitySelector.js

```diff
@@ -0,0 +1,87 @@
+import styled from 'styled-components';
+import useStore from '../hooks/useStore';
+
+export default function QuantitySelector({ product }) {
+  const addQuantity = useStore(state => state.addQuantity);
+  const subtractQuantity = useStore(state => state.subtractQuantity);
+  const removeQuantity = useStore(state => state.removeQuantity);
+
+  return (
+    <QuantityWrapper>
+      <StyledSelector>
+        <StyledQuantityButton
+          type="button"
+          onClick={() => {
+            subtractQuantity(product.id);
+          }}
```

Comment:
```suggestion
          onClick={() => setQuantity(product.id, Math.max(product.quantity - 1, 0))}
```

```suggestion
          onClick={() => subtractQuantity(product.id)}
```

the former is if you decide to use the setQuantity function, the latter if you don't wish to use it.